### PR TITLE
Fixes etcdutl snapshot restore memory alloc. issue

### PR DESCRIPTION
**Pull request by fatihusta**
Fixes: #16052

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: etcdutl/etcdutl/snapshot_command.go [hunk 1]
```diff
@@ -75,6 +76,7 @@ func NewSnapshotRestoreCommand() *cobra.Command {
 	cmd.Flags().StringVar(&restorePeerURLs, "initial-advertise-peer-urls", defaultInitialAdvertisePeerURLs, "List of this member's peer URLs to advertise to the rest of the cluster")
 	cmd.Flags().StringVar(&restoreName, "name", defaultName, "Human-readable name for this member")
 	cmd.Flags().BoolVar(&skipHashCheck, "skip-hash-check", false, "Ignore snapshot integrity hash value (required if copied from data directory)")
+	cmd.Flags().Int64Var(&quotaBackendBytes, "quota-backend-bytes", 0, "Backend quota size. 0 means use the default quota.)")
```
Comment (serathius) [inline etcdutl/etcdutl/snapshot_command.go:79]: Can you replace 0 with the default quota?

File: server/storage/backend/backend.go [hunk 2]
```diff
@@ -151,6 +151,15 @@ type BackendConfig struct {
 	Hooks Hooks
 }
 
+type NewDefBackend struct {
```
Comment (serathius) [inline server/storage/backend/backend.go:154]: No need for struct here.

File: etcdutl/etcdutl/snapshot_command.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fatihusta) [reply etcdutl/etcdutl/snapshot_command.go]: I refactored all PR. I hope better then old PR. Could you please review again.

File: etcdutl/etcdutl/snapshot_command.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ var (
 	restorePeerURLs     string
 	restoreName         string
 	skipHashCheck       bool
+	initialMmapSize   = uint64(10 * 1024 * 1024 * 1024)
```
Comment (tjungblu) [inline etcdutl/etcdutl/snapshot_command.go:41]: 10 gigs? why so large?

Comment (fatihusta) [reply etcdutl/etcdutl/snapshot_command.go:41]: I copied from [here](https://github.com/etcd-io/etcd/blob/da49157b20ffcfce255c2fad06c6e83cf51c7fe8/server/storage/backend/backend.go#L42)

File: server/config/config.go [hunk 4]
```diff
@@ -124,6 +124,7 @@ type ServerConfig struct {
 	CompactionBatchLimit    int
 	CompactionSleepInterval time.Duration
 	QuotaBackendBytes       int64
+	InitialMmapSize       	uint64
```
Comment (jmhbnz) [inline server/config/config.go:127]: Nit - Alignment of `unit64` here looks slight off.

**Review by jmhbnz — COMMENTED** [review]
I think this is looking pretty good, thanks @fatihusta.

Suggest potentially squashing commits down to 1 or 2 if you have time. Also one small nitpick on alignment.

File: server/config/config.go (hunk 4 above)
Comment (fatihusta) [reply server/config/config.go]: Done. :) thanks for your review.

**Review by fuweid — APPROVED** [review]
LGTM if the commits can be squashed so that it can be easy to backport.

**Comment by ivanvc** [comment]
As discussed in the fortnightly etcd triage meeting, I'm continuing the work by squashing the commit and rebasing main in PR #17277.

**Comment by ahrtr** [comment]
superseded by https://github.com/etcd-io/etcd/pull/17277

**Pull request closed without merging**